plugins/inputs/influxdb_listener/influxdb_listener.go

```diff
@@ -0,0 +1,395 @@
+}
+
+func (h *InfluxDBListener) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	h.requestsRecv.Incr(1)
+	h.mux.ServeHTTP(res, req)
+	h.requestsServed.Incr(1)
+}
+
+func (h *InfluxDBListener) handleQuery() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		defer h.queriesServed.Incr(1)
+		// Deliver a dummy response to the query endpoint, as some InfluxDB
+		// clients test endpoint availability with a query
+		res.Header().Set("Content-Type", "application/json")
+		res.Header().Set("X-Influxdb-Version", "1.0")
+		res.WriteHeader(http.StatusOK)
+		res.Write([]byte("{\"results\":[]}"))
+	}
+}
+
+func (h *InfluxDBListener) handlePing() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		defer h.pingsServed.Incr(1)
+		verbose := req.URL.Query().Get("verbose")
+
+		// respond to ping requests
+		if verbose != "" && verbose != "0" && verbose != "false" {
+			res.WriteHeader(http.StatusOK)
+			b, _ := json.Marshal(map[string]string{"version": "1.0"}) // based on header set above
+			res.Write(b)
+		} else {
+			res.WriteHeader(http.StatusNoContent)
+		}
+	}
+}
+
+func (h *InfluxDBListener) handleDefault() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		defer h.notFoundsServed.Incr(1)
+		http.NotFound(res, req)
+	}
+}
+
+func (h *InfluxDBListener) handleWrite() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		defer h.writesServed.Incr(1)
+		// Check that the content length is not too large for us to handle.
+		if req.ContentLength > h.MaxBodySize.Size {
+			tooLarge(res)
+			return
+		}
+
+		db := req.URL.Query().Get("db")
+
+		body := req.Body
+		body = http.MaxBytesReader(res, body, h.MaxBodySize.Size)
+		// Handle gzip request bodies
+		if req.Header.Get("Content-Encoding") == "gzip" {
+			var err error
+			body, err = gzip.NewReader(req.Body)
+			if err != nil {
+				h.log.Debugf("Error decompressing request body: %v", err.Error())
+				badRequest(res, err.Error())
+				return
+			}
+			defer body.Close()
+		}
+
+		parser := influx.NewStreamParser(body)
+		parser.SetTimeFunc(h.timeFunc)
+
+		var err error
+		precisionStr := req.URL.Query().Get("precision")
+		if precisionStr != "" {
+			var precision time.Duration
+			precision = getPrecisionMultiplier(precisionStr)
+			parser.SetTimePrecision(precision)
+		}
+
+		var m telegraf.Metric
+		var parseErrorCount int
+		var lastPos int
+		lastPos = 0
+		var firstParseErrorStr string
+		for {
+			m, err = parser.Next()
+			pos := parser.Position()
+			h.bytesRecv.Incr(int64(pos - lastPos))
+			lastPos = pos
+
+			//continue parsing metrics even if some are malformed
+			if parseErr, ok := err.(*influx.ParseError); ok {
+				parseErrorCount += 1
+				errStr := parseErr.Error()
+				if firstParseErrorStr == "" {
+					firstParseErrorStr = errStr
+				} else {
+				}
+				continue
+			} else if err != nil {
```

Comment:
don't need else, can be a new `if`, or removed if you do the `for err == nil`

Reply:
If I remove the else, it will break out too early when err is a parse error that we've already handled.  If I do ```for err == nil```  and remove the check here, it will add a metric after a non-parse error when the metric isn't valid.  I think the logic is right as is.